### PR TITLE
chore_: cleanup unused cmd

### PR DESCRIPTION
**Pull request by igor-sirotin**
Removed:
- `cmd/bootnode`
- `cmd/spiff-workflow`
- `cmd/node-canary`
- Some outdated/unused files
- Cleanup references to some non-existent `cmd/statusd-prune`

Checking clients:
- https://github.com/status-im/status-desktop/pull/16711
- https://github.com/status-im/status-mobile/pull/21556

### CODE REVIEW

File: MAILSERVER.md [hunk 1]
```diff
@@ -1,75 +0,0 @@
-# Description
-
-A poorly named "Mailserver" is essentially a Whisper node that stores message history in either a LevelDB or PostgreSQL database.
-
-A Status app user can run their own Mailserver for faster message retrieval or additional security.
```
Comment (jakubgs) [inline MAILSERVER.md:5]: Why not just move this ~~and `BOOTNODE.md`~~ to `_docs`?

**Review by jakubgs — CHANGES_REQUESTED** [review]
I don't know if I like the removal of setup in `_assets/compose` and `_assets/systemd` for `mailserver`. It is still quite a valid part of the setup. Mailservers(waku v1 nodes) are still being used in `notify.prod` fleet, so I don't think removing this stuff makes sense.

File: MAILSERVER.md (hunk 1 above)
Comment (igor-sirotin) [reply MAILSERVER.md:5]: Well, `BOOTNODE.md` is about running `cmd/bootnode`, doesn't make sense to keep it if we remove the app.

`MAILSERVER.md` runs `cmd/statusd` (which stays for now), so I suppose we could keep it indeed. Though it basically shouldn't be used at this point. If someone needs to run a Mailserver, it's better to checkout some older version, running anything WakuV1 from current state has no guarantees to be correct.

@jakubgs do we have plans to kick `notify.prod`?

Comment (jakubgs) [reply MAILSERVER.md:5]: As far as I know we can't unless we implement support for Gorush directly in `nim-waku`, and that's not happening anytime soon.

I'm fine with removing `bootnode` stuff.

Comment (richard-ramos) [reply MAILSERVER.md:5]: > @jakubgs: Mailservers(waku v1 nodes) are still being used in notify.prod fleet,

Why is notify.prod using wakuv1? Is this fleet used only by those users that have not updated their clients to more recent versions?

Comment (igor-sirotin) [reply MAILSERVER.md:5]: @jakubgs Yeah I also realised that `notify.prod` is not related to `Mailserver` 🤔 
`Mailserver == Storenode`, I thought it has nothing to do with the push notifications server
And the push notification server is still there, I'm not removing anything related to it.

Comment (jakubgs) [reply MAILSERVER.md:5]: Sorry @richard-ramos, it's actually using V2:
```yaml
status_go_waku_v1_enabled: false
status_go_waku_v2_enabled: true
```
https://github.com/status-im/infra-push-notify/blob/b9a02e093a9de2e028a3d50f1a61015c02a57390/ansible/group_vars/notify.yml#L26-L27

@igor-sirotin you're right, the `push.prod` fleet has no storing enabled:
```yaml
status_go_history_enabled: false
```
https://github.com/status-im/infra-push-notify/blob/b9a02e093a9de2e028a3d50f1a61015c02a57390/ansible/group_vars/notify.yml#L25

So it should be fine.

**Review by jakubgs — APPROVED** [review]

**Review by richard-ramos — APPROVED** [review]

**Comment by igor-sirotin** [comment]
Merging as a non-braking change, because it doesn't affect clients.
And we're most probably about to change the versioning strategy.

**Pull request merged**